Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -395,3 +407,406 @@ lemma completedCosZeta₀_one_sub (a : UnitAddCircle) (s : ℂ) :
+cancelled out by the Gamma factor). -/
+lemma differentiableAt_hurwitzZetaEven (a : UnitAddCircle) {s : ℂ} (hs' : s ≠ 1) :
+    DifferentiableAt ℂ (hurwitzZetaEven a) s := by
+  have := differentiableAt_update_of_residue
+    (fun t ht ht' ↦ differentiableAt_completedHurwitzZetaEven a (Or.inl ht) ht')
+    (completedHurwitzZetaEven_residue_zero a) s hs'
+  simp_rw [div_eq_mul_inv, ite_mul, zero_mul, ← div_eq_mul_inv] at this
+  exact this
+
+lemma hurwitzZetaEven_residue_one (a : UnitAddCircle) :
+    Tendsto (fun s ↦ (s - 1) * hurwitzZetaEven a s) (𝓝[≠] 1) (𝓝 1) := by
+  have : Tendsto (fun s ↦ (s - 1) * completedHurwitzZetaEven a s / Gammaℝ s) (𝓝[≠] 1) (𝓝 1) := by
+    simpa only [Gammaℝ_one, inv_one, mul_one] using (completedHurwitzZetaEven_residue_one a).mul
+      <| (differentiable_Gammaℝ_inv.continuous.tendsto _).mono_left nhdsWithin_le_nhds
+  refine this.congr' ?_
+  filter_upwards [eventually_ne_nhdsWithin one_ne_zero] with s hs
+  simp_rw [hurwitzZetaEven_def_of_ne_or_ne (Or.inr hs), mul_div_assoc]
+
+lemma differentiableAt_hurwitzZetaEven_sub_one_div (a : UnitAddCircle) :
+    DifferentiableAt ℂ (fun s ↦ hurwitzZetaEven a s - 1 / (s - 1) / Gammaℝ s) 1 := by
+  suffices DifferentiableAt ℂ
+      (fun s ↦ completedHurwitzZetaEven a s / Gammaℝ s - 1 / (s - 1) / Gammaℝ s) 1 by
+    apply this.congr_of_eventuallyEq
+    filter_upwards [eventually_ne_nhds one_ne_zero] with x hx
+    rw [hurwitzZetaEven, Function.update_noteq hx]
+  simp_rw [← sub_div, div_eq_mul_inv _ (Gammaℝ _)]
+  refine DifferentiableAt.mul ?_ differentiable_Gammaℝ_inv.differentiableAt
+  simp_rw [completedHurwitzZetaEven_eq, sub_sub, add_assoc]
+  conv => enter [2, s, 2]; rw [← neg_sub, div_neg, neg_add_self, add_zero]
+  exact (differentiable_completedHurwitzZetaEven₀ a _).sub
+    <| (differentiableAt_const _).div differentiableAt_id one_ne_zero
+
+/-- Expression for `hurwitzZetaEven a 1` as a limit. (Mathematically `hurwitzZetaEven a 1` is
+undefined, but our construction assigns some value to it; this lemma is mostly of interest for
+determining what that value is). -/
+lemma tendsto_hurwitzZetaEven_sub_one_div_nhds_one (a : UnitAddCircle) :
+    Tendsto (fun s ↦ hurwitzZetaEven a s - 1 / (s - 1) / Gammaℝ s) (𝓝 1)
+    (𝓝 (hurwitzZetaEven a 1)) := by
+  simpa only [one_div, sub_self, div_zero, Gammaℝ_one, div_one, sub_zero] using
+    (differentiableAt_hurwitzZetaEven_sub_one_div a).continuousAt.tendsto
+
+lemma differentiable_hurwitzZetaEven_sub_hurwitzZetaEven (a b : UnitAddCircle) :
+    Differentiable ℂ (fun s ↦ hurwitzZetaEven a s - hurwitzZetaEven b s) := by
+  intro z
+  rcases ne_or_eq z 1 with hz | rfl
+  · exact (differentiableAt_hurwitzZetaEven a hz).sub (differentiableAt_hurwitzZetaEven b hz)
+  · -- NB. This can be written more tidy with `convert`, but the `convert` version is 3x slower, as
+    -- it spends 877 TC synthesis steps vainly trying to infer `Subsingleton (ℂ → ℂ)`.
```

Comment:
Hopefully this problem will go away once [lean4#3996](https://github.com/leanprover/lean4/issues/3996) has been dealt with.

Reply:
I added a TODO note.

Reply:
With the latest mathlib master, the `convert` is now fast again, so I have reinstated it.